app/src/main/java/com/example/tasky/agenda/data/DateGeneratorImpl.kt

```diff
@@ -0,0 +1,43 @@
+package com.example.tasky.agenda.data
+
+import com.example.tasky.agenda.domain.DateGenerator
+import com.example.tasky.agenda.presentation.model.Day
```

Comment:
You are in the Data layer, but are using stuff from the Presentation. This shouldn't happen

Reply:
So should my UI data classes belong to domain?

Reply:
Nono, the ui data class should be kept in presentation
I explained how to do it on another comment
But let me know if you need more info